website/docs/r/dx_macsec_key.html.markdown

```diff
@@ -0,0 +1,70 @@
+---
+subcategory: "Direct Connect"
+layout: "aws"
+page_title: "AWS: aws_dx_macsec_key"
+description: |-
+  Provides a MAC Security (MACSec) secret key resource for use with Direct Connect.
+---
+
+# Resource: aws_dx_macsec_key
+
+Provides a MAC Security (MACSec) secret key resource for use with Direct Connect. See [MACsec prerequisites](https://docs.aws.amazon.com/directconnect/latest/UserGuide/direct-connect-mac-sec-getting-started.html#mac-sec-prerequisites) for information about MAC Security (MACsec) prerequisites.
+
+Creating this resource will also create a resource of type [`aws_secretsmanager_secret`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/secretsmanager_secret) which is managed by Direct Connect. While you can import this resource into your Terraform state, because this secret is managed by Direct Connect, you will not be able to make any modifications to it. See [How AWS Direct Connect uses AWS Secrets Manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/integrating_how-services-use-secrets_directconnect.html) for details.
+
+~> **Note:** All arguments including `ckn` and `cak` will be stored in the raw state as plain-text.
+[Read more about sensitive data in state](https://www.terraform.io/docs/state/sensitive-data.html).
+
+~> **Note:** The `secret_arn` argument can only be used to reference a previously created MACSec key. You cannot associate a Secrets Manager secret created outside of the `aws_dx_macsec_key` resource.
+
+## Example Usage
+
+### Create MACSec key with CKN and CAK
+
+```terraform
+data "aws_dx_connection" "example" {
+  name = "tf-dx-connection"
+}
+
+resource "aws_dx_macsec_key" "test" {
+  connection_id = data.aws_dx_connection.example.id
+  ckn           = "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"
+  cak           = "abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789"
+}
+```
+
+### Create MACSec key with existing Secrets Manager secret
+
+```terraform
+data "aws_dx_connection" "example" {
+  name = "tf-dx-connection"
+}
+
+data "aws_secretsmanager_secret" "example" {
+  name = "directconnect!prod/us-east-1/directconnect/0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"
+}
+
+resource "aws_dx_macsec_key" "test" {
```

Comment:
Have you considered naming this resource `aws_dx_macsec_key_association`? Since MACsec keys can only be associated/described/dissociated, renaming the resource may help preventing any confusion. 
```suggestion
resource "aws_dx_macsec_key_association" "test" {
```